python/lsst/qa/explorer/functors.py

```diff
@@ -767,3 +793,169 @@ class Seeing(Functor):
     def _func(self, df):
         return 0.168*2.35*np.sqrt(0.5*(df['ext_shapeHSM_HsmPsfMoments_xx']**2 +
                                        df['ext_shapeHSM_HsmPsfMoments_yy']**2))
+
+
+class e1(Functor):
+    name = "Ellipticity e1"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        self._columns = [self.colXX, self.colXY, self.colYY]
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return df[self.colXX] - df[self.colYY] / (df[self.colXX] + df[self.colYY])
+
+
+class e2(Functor):
+    name = "Ellipticity e2"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return 2*df[self.colXY] / (df[self.colXX] + df[self.colYY])
+
+
+class EffectiveRadius(Functor):
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        super().__init__(**kwargs)
+
+    def computeReScalar(xx, yy, xy):
+        quad = afwGeom.Quadrupole(xx, yy, xy)
+        axes = afwGeom.ellipses.Axes(quad)
+        return np.sqrt(axes.getA() * axes.getB())
+
+    computeReVec = np.vectorize(computeReScalar)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        reArr = self.computeReVec(df[self.colXX], df[self.colYY], df[self.colXY])
+        return pd.Series(reArr, index=df.index)
+
+
+class ReferenceBand(Functor):
+    name = 'Reference Band'
+    shortname = 'refBand'
+
+    @property
+    def columns(self):
+        return ["merge_measurement_i",
+                "merge_measurement_r",
+                "merge_measurement_z",
+                "merge_measurement_y",
+                "merge_measurement_g"]
+
+    def _func(self, df):
+        def getFilterAliasName(row):
+            # get column name with the max value (True > False)
+            colName = row.idxmax()
+            return colName.replace('merge_measurement_', '')
+
+        return df[self.columns].apply(getFilterAliasName, axis=1)
+
+
+class Photometry(Functor):
+    AB_FLUX_SCALE = 3.630780547701013425e12  # dn to NanoJansky
```

Comment:
This `dn` is for coadd images with `MAGZERO=27`, I assume?

Reply:
Bad inline comment. Thanks for pointing it out. AB zeropoint is 3631 Janskys.